src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -96,6 +96,34 @@ internal void InitializeCurl()
 
             public void EnsureResponseMessagePublished()
             {
+                // If this is the response hasn't been published yet, do any finaly processing of the response 
+                // message before it's published.
+                if (!Task.IsCompleted)
+                {
+                    // On Windows, if the response was automatically decompressed, Content-Encoding and Content-Length
+                    // headers are removed from the response. Do the same thing here.
+                    DecompressionMethods dm = _handler.AutomaticDecompression;
+                    if (dm != DecompressionMethods.None)
+                    {
+                        HttpContentHeaders contentHeaders = _responseMessage.Content.Headers;
+                        IEnumerable<string> encodings;
+                        if (contentHeaders.TryGetValues(HttpKnownHeaderNames.ContentEncoding, out encodings))
```

Comment:
In addition to getting a header at a time, we also need to support either order. If we get length first, we need to add it to the collection anyway. If we get encoding first, we need to store somewhere that we got it, which means it's not very pay-for-play, as we need that regardless of whether decompression is being used. And that additional state would also need to be reset in cases like redirection, adding more possibility of error. Doing it here is very localized.